### PR TITLE
workflows: Allow urls-check to write issues

### PR DESCRIPTION
**Pull request by martinpitt**
Commit bc7e2bf38f8f729 was wrong: This workflow only doesn't need any permissions if all URLs are valid. But if not, then it wants to update/create an issue to notify us about that.

---

Fixes [this failure](https://github.com/cockpit-project/cockpit/actions/runs/8266231191/job/22613728485) which [started to happen 3 weeks ago](https://github.com/cockpit-project/cockpit/actions/workflows/urls-check.yml).

### CODE REVIEW

**Comment by martinpitt** [comment]
The [manual run against this branch](https://github.com/cockpit-project/cockpit/actions/runs/8266365176), and it passed again. It filed #20167

**Comment by martinpitt** [comment]
#20167 is bogus. I added another commit here to get some more data, and [triggered another run](https://github.com/cockpit-project/cockpit/actions/runs/8266711073)

**Comment by martinpitt** [comment]
OK, that must have been a transient network issue then .. the last check was ok. To test the second commit, I deliberately damaged that URL in the source (`/sites` → `/stes`), and got:
```
Checking: https://www.dmtf.org/stes/default/files/standards/documents/DSP0134_2.7.1.pdf
Checked 177 URLs out of which 1 is/are invalid:
    https://www.dmtf.org/stes/default/files/standards/documents/DSP0134_2.7.1.pdf: 404 Not Found
```

File: tools/urls-check [hunk 1]
```diff
@@ -123,9 +123,9 @@ def check_urls(verbose):
             if resp.geturl() != url and not any(fnmatch.fnmatch(url, pattern) for pattern in KNOWN_REDIRECTS):
                 redirects.append(url)
             if resp.getcode() >= 400:
-                failed.append(url)
-        except urllib.error.URLError:
-            failed.append(url)
+                failed.append(f"{url} : {resp.getcode()} {resp.reason}")
```
Comment (martinpitt) [inline tools/urls-check:126]: As this looks a bit unusual/asymmetric, I tested this code path as well, by changing the 400 to 200, and then got:
```
Checked 177 URLs out of which 1 is/are invalid:
    https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_2.7.1.pdf: 200 OK
```

**Review by allisonkarlitskaya — APPROVED** [review]

**Pull request merged**